src/EducationService.Data/UserEducationRepository.cs

```diff
@@ -46,5 +47,29 @@ public async Task DisactivateEducationsAsync(Guid userId, Guid modifiedBy)
 
       await _provider.SaveAsync();
     }
+
+    public async Task<List<DbUserEducation>> FindAsync(FindUsersFilter filter)
+    {
+      IQueryable<DbUserEducation> query = _provider.UsersEducations.AsQueryable();
+
+      query = query.Where(e => e.UserId == filter.UserId);
+
+      if (filter.EducationFormId.HasValue)
+      {
+        query = query.Where(e => e.EducationForm.Id == filter.EducationFormId.Value);
+      }
+
+      if (filter.EducationTypeId.HasValue)
+      {
+        query = query.Where(e => e.EducationType.Id == filter.EducationTypeId.Value);
+      }
+
+      if (filter.Completeness.HasValue)
+      {
+        query = query.Where(e => e.Completeness == (int)filter.Completeness.Value);
+      }
+
+      return await query.Skip(filter.SkipCount).Take(filter.TakeCount).ToListAsync();
```

Comment:
remove await and async, return Task
you`re already using await in command on 48 line

Reply:
Shouldn't the method working with the database be asynchronous?

Reply:
Done